misc/python/materialize/scratch.py

```diff
@@ -438,11 +323,35 @@ def is_old(i: InstanceTypeDef) -> bool:
         if delete_after is None:
             return False
         delete_after = float(delete_after)
-        return datetime.now(timezone.utc).timestamp() > delete_after
+        return datetime.datetime.utcnow().timestamp() > delete_after
 
     return [
         i
         for r in boto3.client("ec2").describe_instances()["Reservations"]
         for i in r["Instances"]
         if is_running(i) and is_old(i)
     ]
+
+
+def mssh(
+    instance: str, command: str, *, stdin: Union[None, int, IO[bytes], bytes] = None
+) -> None:
+    """Runs a command over SSH via EC2 Instance Connect."""
+    if command:
+        # The actual invocation of SSH becomes unreadable quoted garbage, so we do
+        # our own printing here.
+        print(f"ubuntu@{instance}$ {command}", file=sys.stderr)
+        # Eval and quote to work around
+        # https://github.com/aws/aws-ec2-instance-connect-cli/pull/25.
+        command = f"eval {shlex.quote(command)}"
+    else:
+        print(f"$ mssh ubuntu@{instance}")
+    spawn.runv(
+        [
+            *SSH_COMMAND,
+            f"ubuntu@{instance}",
+            command,
+        ],
+        stdin=stdin,
+        print_to=open(os.devnull, "w"),
```

Comment:
why no output?

Reply:
There's a comment at the top of the function that explains the situation. I reworded that comment a bit to make it more clear that it applies to the whole function. `print_to` doesn't control the output of the command itself; just where the `> command args` header goes. That header is unreadable garbage thanks to the shell quoting, so we suppress it.